spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/ComposedTaskNode.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core.dsl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.cloud.dataflow.core.dsl.graph.Graph;
+
+/**
+ * The root AST node for any AST parsed from a composed task specification.
+ *
+ * Andy Clement
+ */
+public class ComposedTaskNode extends AstNode {
+
+	/**
+	 * The DSL text that was parsed to create this ComposedTaskNode.
+	 */
+	private String composedTaskText;
+
+	/**
+	 * The sequence of LabelledNodes parsed from the specification.
+	 */
+	private List<LabelledComposedTaskNode> sequences;
+
+	/**
+	 * All the apps mentioned in the composed task definition.
+	 */
+	private Set<String> taskApps;
+
+	public ComposedTaskNode(String composedTaskSpecification, List<LabelledComposedTaskNode> sequences) {
+		super((sequences.size()==0) ? 0 : sequences.get(0).getStartPos(), 
```

Comment:
I can appreciate on public 'api' you want to assert things. So does this become more a question of limiting visibility where appropriate. There is no reason I can see for this to be public, strictly speaking. So rather than assert I'd reduce visibility.